supplementary/alica_tracing/include/tracing/Trace.h

```diff
@@ -56,17 +51,21 @@ class Trace : public alica::IAlicaTrace
     //        {"wait.time", "waited 10 sec for agent 2"}});
     // BE CAREFUL about what you want to log,
     // These are supposed to be micro logs to be carried over the network.
-    void setLog(std::pair<std::string, std::string> logEntry);
+    void setLog(std::pair<std::string, std::string> logEntry) override;
 
     // When operation being traced by this instance fail, call this api
     // Error traces are highlighted in tracing ui.
-    void markError(const std::string& description);
+    void markError(const std::string& description) override;
+
+    // Explicitly set the trace as finished. Any calls to setTag, setLog & markError after this call leaves
+    // the trace in a valid by unspecified state. Calling context on a finished trace is a valid operation
+    void finish() override;
 
     // Get the context of this trace to propogate across process boundary
-    std::string context() const;
+    std::string context() const override;
 
 private:
-    void finish();
+    void setTag(const std::string& key, const RawTraceValue& value);
```

Comment:
docs

Reply:
this is an internal method, so I didn't put docs

Reply:
I'd still add docs, for others devs from DIS that have to implement things here :P

Reply:
the API setTag is already documented in the public interface of this class